### PR TITLE
[XCConfigHelper] Adds $(inherited) to generated OTHER_SWIFT_FLAGS

### PR DESCRIPTION
**Pull request by danthorpe**
To address (I think) a remaining issue in #2983.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -108,6 +108,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Samuel Giddins](https://github.com/segiddins)
   [#2983](https://github.com/CocoaPods/CocoaPods/issues/2983)
 
+* Fix the `OTHER_SWIFT_FLAGS` generated, so it inherits previous definitions.
```
Comment (segiddins) [inline CHANGELOG.md]: two trailing spaces, please

**Comment by segiddins** [comment]
:+1: once my one comment has been address. I'll rebuild the integration specs upon merge

**Comment by danthorpe** [comment]
Okay, added the trailing spaces. So you're aware before you merge this, I've not actually re-build cocoapods with this change to see if it works - kinda going on faith that `OTHER_SWIFT_FLAGS` respects `$(inherited)`.

**Comment by segiddins** [comment]
Manually merged. Thanks @danthorpe !

**Pull request closed without merging**